community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf

```diff
@@ -25,7 +25,7 @@ locals {
     }
   ]
 
-  service_account_email = coalesce(var.service_account_email, data.google_compute_default_service_account.default.email)
+  service_account_email = coalesce(var.service_account_email, data.google_compute_default_service_account.default[0].email)
```

Comment:
Please test with non-empty `var.service_account_email`, to make sure that `coalesce` does "short-circuiting" and will not fail with "index out of bounds"

Reply:
Good catch. It did not exit the statement and failed with an index error when `var.service_account_email` was non-empty. I changed it to an if statement that checks for null or empty string instead of using `coalesce`.